fuzz/fuzz_targets/sha256.rs

```diff
@@ -0,0 +1,33 @@
+#![no_main]
+
+use arbitrary::{Arbitrary, Result, Unstructured};
+use libfuzzer_sys::fuzz_target;
+
+use bitvm::hash::sha256::{reference_sha256, test_sha256_with};
+use bitvm::hash::sha256_u4::test_sha256_u4_with;
+use bitvm::hash::sha256_u4_stack::test_sha256_u4_stack_with;
+
+#[derive(Debug)]
+struct LimitedBytes(Vec<u8>);
+
+impl<'a> Arbitrary<'a> for LimitedBytes {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        let size = u.int_in_range(0..=512)?;
```

Comment:
There is an off-by-one error here. The implementation will panic if the input length is 512 bytes.